@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -137,6 +137,38 @@ const MAX_CORNER_ANGLE = Math.PI / 8;
  */
 const INVALID_ARRAY_INDEX = -1;
 
+function maxValue(array: ArrayLike<number>): number {
+    let max = -Infinity;
+
+    // tslint:disable-next-line: prefer-for-of
+    for (let i = 0; i < array.length; ++i) {
+        if (array[i] > max) {
+            max = array[i];
+        }
+    }
+
+    return max;
+}
+
+function createIndexBuffer(elements: ArrayLike<number>) {
+    return maxValue(elements) > 65535 ? new Uint32Array(elements) : new Uint16Array(elements);
+}
+
+function createIndexBufferAttribute(
+    elements: ArrayLike<number>,
+    name: string = "index"
+): BufferAttribute {
+    const type = maxValue(elements) > 65535 ? "uint32" : "uint16";
```

Comment:
could you pass in an optional value which has the vertices.length (if available), if it is less than 65k, then you can avoid the maxValue function call?

Reply:
it's not vertexLength, it's the max value of the array.

Reply:
But shouldn't the highest value of the index buffer equal(or be smaller than) `posAttr.count - 1`?

Reply:
why isn't it vertextLength? (well actually vertextLength-1)

Reply:
sure, using vertex.length may be OK with the way we create geometries right now.